presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergMetadataListing.java

```diff
@@ -0,0 +1,197 @@
+import io.prestosql.tests.DistributedQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static io.prestosql.connector.informationschema.InformationSchemaMetadata.INFORMATION_SCHEMA;
+import static io.prestosql.connector.informationschema.InformationSchemaMetadata.TABLE_COLUMNS;
+import static io.prestosql.spi.security.SelectedRole.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+
+public class TestIcebergMetadataListing
+        extends AbstractTestQueryFramework
+{
+    private static final String ICEBERG_CATALOG = "iceberg";
+    private static final String HIVE_CATALOG = "hive";
+    private static final String TEST_SCHEMA = "test_schema";
+    private static final String ICEBERG_TABLE1 = "iceberg_table1";
+    private static final String ICEBERG_TABLE2 = "iceberg_table2";
+    private static final String HIVE_TABLE = "hive_table";
+
+    public TestIcebergMetadataListing()
+    {
+        super(TestIcebergMetadataListing::createQueryRunner);
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        @Language("SQL") String createSchema = "CREATE SCHEMA "
+                + HIVE_CATALOG + "." + TEST_SCHEMA;
+
+        @Language("SQL") String createIcebergTable1 = "CREATE TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE1
+                + "("
+                + "  _string VARCHAR"
+                + ", _integer INTEGER"
+                + ")";
+
+        @Language("SQL") String createIcebergTable2 = "CREATE TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE2
+                + "("
+                + "  _string VARCHAR"
+                + ", _integer INTEGER"
+                + ", _double DOUBLE"
+                + ", _boolean BOOLEAN"
+                + ") "
+                + "WITH ("
+                + "partitioning = ARRAY["
+                + "  '_string'"
+                + ", '_integer'])";
+
+        @Language("SQL") String createHiveTable = "CREATE TABLE "
+                + HIVE_CATALOG + "." + TEST_SCHEMA + "." + HIVE_TABLE
+                + "("
+                + "  _string VARCHAR"
+                + ", _integer INTEGER"
+                + ", _double DOUBLE"
+                + ")";
+
+        assertQuerySucceeds(createSchema);
+        assertQuerySucceeds(createIcebergTable1);
+        assertQuerySucceeds(createIcebergTable2);
+        assertQuerySucceeds(createHiveTable);
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        @Language("SQL") String dropSchema = "DROP SCHEMA "
+                + HIVE_CATALOG + "." + TEST_SCHEMA;
+
+        @Language("SQL") String dropIcebergTable1 = "DROP TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE1;
```

Comment:
Use `IF EXISTS` when dropping. Otherwise tearDown won't clean it up when setUp fails in the middle.